server/routes/ad.ts

```diff
@@ -302,10 +303,7 @@ const searchDetector = async (
       },
     };
   } catch (err) {
-    if (
-      err.statusCode === 404 &&
-      get<string>(err, 'body.error.type', '') === 'index_not_found_exception'
-    ) {
+    if (isIndexNotFoundError(err)) {
```

Comment:
Just confirm, this method `searchDetector ` only used on dashboard and detector list page? The index not found error is caused by detector index not created yet and this error is not visible to user?

Reply:
> Just confirm, this method `searchDetector ` only used on dashboard and detector list page? The index not found error is caused by detector index not created yet and this error is visible to user?

so `searchDetector` is only called in the create detector page, and `getDetectors` is only called on the dashboard and detector list pages. In all of these cases, we don't want this error to be visible to the user since it only means the index hasn't been created yet, which isn't a problem.

I'm not changing any functionality in `searchDetector`, but instead just encapsulating the logic in a util function so it can be used by multiple functions

Reply:
Cool, just make sure the change will not break current design. For example, some page may need to surface the index not found error.